DFC.ServiceTaxonomy.GraphSync/Activities/AuditSyncIssuesTask.cs

```diff
@@ -0,0 +1,144 @@
+            _partSyncers = partSyncers.ToDictionary(x => x.PartName ?? "Eponymous");
+            _syncFailedContentItems = new List<ContentItem>();
+            _contentTypes = contentDefinitionManager
+                .ListTypeDefinitions()
+                .Where(x => x.Parts.Any(p => p.Name == "GraphSyncPart"))
+                .ToDictionary(x => x.Name);
+        }
+
+        private readonly ISession _session;
+        private readonly IGraphDatabase _graphDatabase;
+        private readonly Dictionary<string, IContentPartGraphSyncer> _partSyncers;
+        private readonly IMergeGraphSyncer _mergeGraphSyncer;
+        private IStringLocalizer T { get; }
+
+        public override string Name => nameof(AuditSyncIssuesTask);
+        public override LocalizedString DisplayText => T["Identify sync issues and retry"];
+        public override LocalizedString Category => T["National Careers Service"];
+
+        private readonly List<ContentItem> _syncFailedContentItems;
+        private readonly Dictionary<string, ContentTypeDefinition> _contentTypes;
+
+        public override IEnumerable<Outcome> GetPossibleOutcomes(WorkflowExecutionContext workflowContext, ActivityContext activityContext)
+        {
+#pragma warning disable S3220 // Method calls should not resolve ambiguously to overloads with "params"
+            return Outcomes(T["Done"]);
+#pragma warning restore S3220 // Method calls should not resolve ambiguously to overloads with "params"
+        }
+
+        public override async Task<ActivityExecutionResult> ExecuteAsync(WorkflowExecutionContext workflowContext, ActivityContext activityContext)
+        {
+            var timestamp = DateTime.UtcNow;
+            var log = (await _session.Query<AuditSyncLog>().FirstOrDefaultAsync()) ??
+                      new AuditSyncLog() {LastSynced = DateTime.MinValue};
+
+            var contentItems = await _session
+                //do we only care about the latest published items?
+                .Query<ContentItem, ContentItemIndex>(x =>
+                    x.ContentType.IsIn(_contentTypes.Keys) && x.Latest && x.Published && (x.CreatedUtc >= log.LastSynced ||  x.ModifiedUtc >= log.LastSynced))
+                .ListAsync();
+
+            foreach (var contentItem in contentItems)
+            {
+                await CheckIfContentItemSynced(contentItem);
+            }
+
+            foreach (var contentItem in _syncFailedContentItems)
+            {
+                try
+                {
+                    await _mergeGraphSyncer.SyncToGraph(contentItem.ContentType, contentItem.Content,
+                        contentItem.CreatedUtc, contentItem.ModifiedUtc);
+
+                    _syncFailedContentItems.Remove(contentItem);
+
+                    await CheckIfContentItemSynced(contentItem);
+                }
+                catch
+                {
+                    //I don't want to do anything here, why won't this build without this comment?
```

Comment:
We should log the details of any exception.
We should also probably log all the content items that require a resync.